include/pros/adi.hpp

```diff
@@ -446,15 +561,15 @@ class ADIUltrasonic : private ADIPort {
 	 * \return The distance to the nearest object in m^-4 (10000 indicates 1
 	 * meter), measured from the sensor's mounting points.
 	 */
-	using ADIPort::get_value;
+	std::int32_t get_value(void) const;
```

Comment:
This doesn’t work because in order for the override command to compile, the original function has to be labeled as virtual.  If you label the original as virtual, than gyro::get_value also tries to override it ev,en though it has a different return type.  So, I don’t think there is a way of doing this, without sacrificing uniformity in function names, and it only adds slight maintainability.

Reply:
Humm, this isn't really ideal. Since you have a virtual destructor, you already have the overhead of the vtable, so there is no good reason not to make the methods virtual. They should be virtual anyways, because if you have an `ADIUltrasonic` that is being treated as a `ADIPort`, you want it to properly dispatch to the derived method. 
However, I see the problem with clashing with `ADIGyro`. Unfortunately, how it is right now is not ideal. The `ADIGyro` simply shadows and hides the derived method, which leads to problems with polymorphism and unexpected issues with the return type changing. I was worried it was going to lead to ambiguity, but at least this passes:
```cpp
pros::ADIGyro g('A');
auto x = g.get_value();
static_assert(std::is_same_v<decltype(x), double>);

pros::ADIAnalogIn i('A');
auto j = i.get_value();
static_assert(std::is_same_v<decltype(j), std::int32_t>);
```
But you can see how it is not ideal to have the return type change, it could lead to unexpected conversions and narrowing. 

What would be ideal is to have proper overriding and matching return type. I'm not sure how that would be done though, so I might have to agree with your conclusion. 
Maybe have the base `get_value` override return integers and have a `get_value_HR` for the 10th precision, idk. If you and others don't think that's a good idea that's fine.

Reply:
The inheritence is private, so I don't think this is an issue.

BTW, my understanding is that most of the cost is in the function call. The vtable data structure has some size overhead, but that wouldn't be the part I worry about. If course, neither matter here.